core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -636,7 +636,9 @@ private[spark] object SparkConf extends Logging {
       DeprecatedConfig("spark.blacklist.killBlacklistedExecutors", "3.1.0",
         "Please use spark.excludeOnFailure.killExcludedExecutors"),
       DeprecatedConfig("spark.yarn.blacklist.executor.launch.blacklisting.enabled", "3.1.0",
-        "Please use spark.yarn.executor.launch.excludeOnFailure.enabled")
+        "Please use spark.yarn.executor.launch.excludeOnFailure.enabled"),
+      DeprecatedConfig("spark.kubernetes.memoryOverheadFactor", "3.3.0",
```

Comment:
This seems wrong and inconsistent with this PR because this PR uses `spark.kubernetes.memoryOverheadFactor` to override the driver or executor specific configurations.
```
  private val memoryOverheadFactor = conf.get(MEMORY_OVERHEAD_FACTOR)
     .getOrElse(conf.get(DRIVER_MEMORY_OVERHEAD_FACTOR))
```

Reply:
Yeah the idea was if you still want to use that we will, otherwise we'll look for the new thing, I thought that was how it was normally done, but I can change it if I got that wrong. Not sure how to handle it since the new config has a default value. "Use the new config if it's specifically passed, otherwise try the old config, and if that's not there use the default value for the new config". 

Wasn't sure exactly how to handle the fact that k8s already had this as its own setting but the other didn't.

Reply:
Checking for new config now before the old config